### PR TITLE
Smoke test for portal release testing

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Ref task: https://mattermost.atlassian.net/browse/MM-52534

### CODE REVIEW

File: data/test-by-folder.json [hunk 1]
```diff
@@ -1848,6 +1848,8 @@
       "Cloud signup — Password requirements",
       "Cloud | Alternative Payment | Verify the UI on payment page when user provides inputs for alternative payment",
       "Delinquency | Compelling Admin to update the to previous paid plan or star on freemium",
+      "Expand License | Should be able to expand from Portal with right start & end dates",
+      "Expand License | Should not show annaul deprication banner to selected clients",
```
Comment (lindalumitchell) [inline data/test-by-folder.json:1852]:
```suggestion
      "Expand License | Should not show annual deprication banner to selected clients",
```

File: data/test-cases/cloud/smoke-t-1.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+**Step 1**
+
+1. Create new account on test portal (https://portal.test.cloud.mattermost.com/, for sake of testing)
+2. Purchase a license and verify on `console/subscriptions` the `Purchase additional seats` button is displayed. Click on this button.
+3. Enter Seats in the RHS
+4. Verify the cost calculated is correct (ex: prorated calculation $7.5 per month x 11 months)
+5. Expand the subscription by purchasing additional seats
+6. Verify the purchase is successful and update seat count is displayed on `console/subscriptions`
+7. Navigate to `console/purchase-history` and verify the new purchase details are listed on the top
+8. Click on `Download invoice` verify the PDF contains right number of Seats, Amount, Date and Time of purchase
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-1.md:46]:
```suggestion
8. Click on `Download invoice` and verify the PDF contains right number of Seats, Amount, Date and Time of purchase
```

File: data/test-cases/cloud/smoke-t-1.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Extra manual step
+
+9. Navigate back to `console/subscriptions` and download the license
+10. Upload license to any self-hosted instance and verify the license details are correctly displayed on the System Console > License page.
+11. Verify purchase email is received by the Admin who purchased additional license
+12. Verify the Seats, Amount, Date and Time of purchase, is in-line with details displayed in the portal
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-1.md:53]:
```suggestion
12. Verify the Seats, Amount, Date and Time of purchase, matches details displayed in the portal
```

File: data/test-cases/cloud/smoke-t-1.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+13. Download the attachment from email and upload the license to mattermost instance
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-1.md:54]:
```suggestion
13. Download the attachment from email and upload the license to Mattermost instance
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+---
+# (Required) Ensure all values are filled up
+name: "Expand License | Should not show annaul deprication banner to selected clients"
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:3]:
```suggestion
name: "Expand License | Should not show annual deprication banner to selected clients"
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+case_hashed: null
+steps_hashed: null
+---
+
+ref defect: <https://mattermost.atlassian.net/browse/MM-52158>
+
+**Step 1**
+
+1. Create new account on test portal (https://portal.test.cloud.mattermost.com/cloud/signup, for sake of testing)
+2. Purchase a `Professional`.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:40]: Should this say "`Professional` license"? Or maybe "`Professional` subscription"?

File: data/test-cases/cloud/smoke-t-2.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+3. Log into stripe account. Switch to Test mode. Search for the customer that just signed up.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:41]:
```suggestion
3. Log into Stripe account. Switch to Test mode. Search for the customer that just signed up.
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+4. Get into subscriptions detail page and update the subscription to become monthly (add a product that has monthly price and update the product ID in the subscription meta-data )
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:42]:
```suggestion
4. Get into subscriptions detail page and update the subscription to become monthly (add a product that has monthly price and update the product ID in the subscription metadata)
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+5. Verify on the mm instance that User can see global banners suggesting to move to Annual subscription
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:43]:
```suggestion
5. Verify on the Mattermost instance that the user can see global banners suggesting to move to Annual subscription
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+6. On stripe, add a metadata `billing-type: internal` or `billing-type: licensed`. and wait for 5 mins.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:44]:
```suggestion
6. On Stripe, add a metadata `billing-type: internal` or `billing-type: licensed`, and wait for 5 mins.
```

File: data/test-cases/cloud/smoke-t-2.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+7. Reload the mm instance and Verify the banner is not visible anymore
```
Comment (lindalumitchell) [inline data/test-cases/cloud/smoke-t-2.md:45]:
```suggestion
7. Reload the Mattermost instance and verify the banner is not visible anymore.
```

**Review by lindalumitchell — APPROVED** [review]
Just added some minor suggestions. :)

File: data/test-cases/cloud/smoke-t-2.md (hunk 6 above)
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/smoke-t-2.md:40]: the button will have only `Professional` text, I'll add 'subscription'

**Pull request merged**